Common/cpp/SharedItems/ShareableValue.cpp

```diff
@@ -20,6 +41,23 @@ void ShareableValue::adaptCache(jsi::Runtime &rt, const jsi::Value &value) {
 }
 
 void ShareableValue::adapt(jsi::Runtime &rt, const jsi::Value &value, ValueType objectType) {
+  bool isRNRuntime = !(module->isUIRuntime(rt));
+  if (value.isObject()) {
+    jsi::Object object = value.asObject(rt);
+    if (!(object.getProperty(rt, HIDDEN_HOST_OBJECT_PROP).isUndefined())) {
+      jsi::Object hiddenProperty = object.getProperty(rt, HIDDEN_HOST_OBJECT_PROP).asObject(rt);
+      if (hiddenProperty.isHostObject<FrozenObject>(rt)) {
+        type = ValueType::ObjectType;
+        if (object.hasProperty(rt, "__worklet") && object.isFunction(rt)) {
+          type = ValueType::WorkletFunctionType;
+        }
+        frozenObject = hiddenProperty.getHostObject<FrozenObject>(rt);
+        adaptCache(rt, value);
+        return;
+      }
+    }
```

Comment:
Why not deal with the case when value is object here entirely? The flow reaches the same condition in line 77(doing nothing between here and there)

Reply:
Because it can be mutable.